### PR TITLE
🔧 Make env picklable

### PR DESCRIPTION
**Pull request by MathisFederico**
# Description

I needed envs to be picklable for using them in multiple process.
Thus I fixed the issue that happens to be lambda functions used for missions generation.

Fixes to fact that envs were not picklable.

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

# Checklist:

- [X] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (None?)
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by rodrigodelazcano** [comment]
Thank you for the PR @MathisFederico. I suggest a few changes. I left a comment for the testing function, and could you update the docs for the [MissionSpace](https://github.com/Farama-Foundation/Minigrid/blob/c13e0ef281ce24c7a6ccb7800244009974b374a6/minigrid/core/mission.py#L12) class (use static method function instead of lambda function).

**Comment by MathisFederico** [comment]
> I suggest a few changes.

Sure ! Where can I find your suggestions ? I don't think you published your review if you have done one !

> could you update the docs for the [MissionSpace](https://github.com/Farama-Foundation/Minigrid/blob/c13e0ef281ce24c7a6ccb7800244009974b374a6/minigrid/core/mission.py#L12) class

Done in last commit !

File: tests/test_envs.py [hunk 1]
```diff
@@ -162,6 +163,13 @@ def test_max_steps_argument(env_spec):
     env.close()
 
 
+@pytest.mark.parametrize("env_id", [spec.id for spec in all_testing_env_specs])
```
[the remark this answers is not in the document]
Comment (rodrigodelazcano) [reply tests/test_envs.py:166]: @MathisFederico  test suggestion :)

**Comment by rodrigodelazcano** [comment]
Thanks! I just tagged you on it, under `tests/test_envs.py`

File: minigrid/minigrid_env.py [hunk 2]
```diff
@@ -87,14 +87,19 @@ class MissionSpace(spaces.Space[str]):
     r"""A space representing a mission for the Gym-Minigrid environments.
```
Comment (rodrigodelazcano) [inline minigrid/minigrid_env.py:87]: If you can change `Gym-Minigrid` to `Minigrid` that will be great

**Comment by MathisFederico** [comment]
> Thanks! I just tagged you on it, under `tests/test_envs.py`

Not sure where to see the tag ... I guess it's on GitHub as I don't see any new commit from your part, but I can't seem to find where you tagged me ^^'

**Review by rodrigodelazcano — CHANGES_REQUESTED** [review]

**Comment by rodrigodelazcano** [comment]
@MathisFederico can you see the review now?

**Comment by MathisFederico** [comment]
Yes ! Sorry I did not have time for now, I may do this tonight if I can find the time !

**Comment by MathisFederico** [comment]
Done ! @rodrigodelazcano

**Pull request merged**